metadata-ingestion/src/datahub/ingestion/source/hex/hex.py

```diff
@@ -0,0 +1,162 @@
+from datahub.ingestion.source.hex.model import Component, Project
+from datahub.ingestion.source.state.stale_entity_removal_handler import (
+    StaleEntityRemovalHandler,
+    StaleEntityRemovalSourceReport,
+    StatefulStaleMetadataRemovalConfig,
+)
+from datahub.ingestion.source.state.stateful_ingestion_base import (
+    StatefulIngestionConfigBase,
+    StatefulIngestionReport,
+    StatefulIngestionSourceBase,
+)
+
+
+class HexSourceConfig(StatefulIngestionConfigBase, PlatformInstanceConfigMixin):
+    workspace_name: str = Field(
+        description="Hex workspace name. You can find this name in your Hex home page URL: https://app.hex.tech/<workspace_name>",
+    )
+    token: SecretStr = Field(
+        description="Hex API token; either PAT or Workflow token - https://learn.hex.tech/docs/api/api-overview#authentication",
+    )
+    base_url: str = Field(
+        default=HEX_API_BASE_URL_DEFAULT,
+        description="Hex API base URL. For most Hex users, this will be https://app.hex.tech/api/v1. "
+        "Single-tenant app users should replace this with the URL they use to access Hex.",
+    )
+    stateful_ingestion: Optional[StatefulStaleMetadataRemovalConfig] = Field(
+        default=None,
+        description="Configuration for stateful ingestion and stale metadata removal.",
+    )
+    include_components: bool = Field(
+        default=True,
+        desciption="Include Hex Components in the ingestion",
+    )
+    page_size: int = Field(
+        default=HEX_API_PAGE_SIZE_DEFAULT,
+        description="Number of items to fetch per Hex API call.",
+    )
+    patch_metadata: bool = Field(
+        default=False,
+        description="Emit metadata as patch events",
+    )
+    collections_as_tags: bool = Field(
+        default=True,
+        description="Emit Hex Collections as tags",
+    )
+    status_as_tag: bool = Field(
+        default=True,
+        description="Emit Hex Status as tags",
+    )
+    categories_as_tags: bool = Field(
+        default=True,
+        description="Emit Hex Category as tags",
+    )
+    project_title_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex pattern for project titles to filter in ingestion.",
+    )
+    component_title_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex pattern for component titles to filter in ingestion.",
+    )
+    set_ownership_from_email: bool = Field(
+        default=True,
+        description="Set ownership identity from owner/creator email",
+    )
+
+
+class HexReport(StaleEntityRemovalSourceReport, HexApiReport):
+    pass
+
+
+@platform_name("Hex")
+@config_class(HexSourceConfig)
+@support_status(SupportStatus.TESTING)
+@capability(SourceCapability.DESCRIPTIONS, "Supported by default")
+@capability(SourceCapability.OWNERSHIP, "Supported by default")
+@capability(SourceCapability.PLATFORM_INSTANCE, "Enabled by default")
```

Comment:
should we say that it's tied to the workspace name, which is already globally unique

Reply:
Not sure what you mean platform instance "tied" to the workspace name

Platform instance is optional and just another decorator that users can add. It may be useful in case of single-tenant hex users.

> For most Hex users, this will be https://app.hex.tech/api/v1. Single-tenant app users should replace the app.hex.tech part of this URL with the URL they use to access Hex (e.g. atreides.hex.tech)

Feel free to make a code suggestion, if you have one.